File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        // 06.12.23 POTD

        int money=0, mon=1, curr=1;
        for(int i=1; i<=n; i++){
            money += curr;
            
            if(i%7==0){
                curr = mon+1;
                mon++;
            } else {
                curr++;
            }
        }

        return money;
    }
};

File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        # 25.10.25 POTD
        ans = 0
        nn = n // 7
        rem = n % 7
        a = 28
        d = 7
        
        # AP for complete weeks
        ans = (nn * (2 * a + (nn-1) * d)) // 2

        # AP for days in last week
        ans += (rem * (2 * (1+nn) + rem - 1)) // 2

        return ans